pkg/translator/prometheusremotewrite/metrics_to_prw.go

```diff
@@ -25,9 +27,33 @@ type Settings struct {
 }
 
 // FromMetrics converts pmetric.Metrics to Prometheus remote write format.
-func FromMetrics(md pmetric.Metrics, settings Settings) (tsMap map[string]*prompb.TimeSeries, errs error) {
-	tsMap = make(map[string]*prompb.TimeSeries)
+func FromMetrics(md pmetric.Metrics, settings Settings) (map[string]*prompb.TimeSeries, error) {
+	c := newPrometheusConverter()
+	errs := c.FromMetrics(md, settings)
+	tss := c.TimeSeries()
+	out := make(map[string]*prompb.TimeSeries, len(tss))
+	for i := range tss {
+		out[strconv.Itoa(i)] = &tss[i]
+	}
+
+	return out, errs
+}
+
+// prometheusConverter converts from OTel write format to Prometheus write format.
+type prometheusConverter struct {
+	unique    map[uint64]*prompb.TimeSeries
+	conflicts map[uint64][]*prompb.TimeSeries
+}
 
+func newPrometheusConverter() *prometheusConverter {
+	return &prometheusConverter{
+		unique:    map[uint64]*prompb.TimeSeries{},
+		conflicts: map[uint64][]*prompb.TimeSeries{},
+	}
+}
+
+// FromMetrics converts pmetric.Metrics to Prometheus remote write format.
+func (c *prometheusConverter) FromMetrics(md pmetric.Metrics, settings Settings) (errs error) {
```

Comment:
```suggestion
func (c *prometheusConverter) fromMetrics(md pmetric.Metrics, settings Settings) errs error {
```

This type isn't exported and isn't implementing an interface (and wouldn't be instantiable if it did) so it shouldn't have exported methods.